bpf-map-pressure-exporter/src/fetcher.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"errors"
+	"os"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/cybozu-go/log"
+)
+
+type IBpfMapPressureFetcher interface {
+	Fetch() []bpfMapPressure
+}
+
+type bpfMapPressureFetcher struct {
+	mapNameStrings []string
+}
+
+type bpfMapPressure struct {
+	mapId       uint32
+	mapName     string
+	mapPressure float64
+}
+
+func newFetcher(mapNameStrings []string) *bpfMapPressureFetcher {
+	return &bpfMapPressureFetcher{
+		mapNameStrings: mapNameStrings,
+	}
+}
+
+func (f *bpfMapPressureFetcher) Fetch() []bpfMapPressure {
+	results := []bpfMapPressure{}
+	var id ebpf.MapID = 0
+	var err error
+	for {
+		id, err = ebpf.MapGetNextID(id)
+		if errors.Is(err, os.ErrNotExist) {
+			break
+		}
+		if err != nil {
+			_ = logger.Warn("failed to get next map id", map[string]interface{}{
+				"id":        id,
+				log.FnError: err,
+			})
+			return nil
```

Comment:
How about returning `results` because the results already calculated should not be a problem.

Reply:
https://github.com/cybozu/neco-containers/pull/1155/commits/7f125beba4c58a82c05741e9bb42ad460474acac